gemini/sample-apps/genwealth/function-scripts/process-pdf/main.py

```diff
@@ -0,0 +1,328 @@
+
+    if metadata.state != documentai.BatchProcessMetadata.State.SUCCEEDED:
+        raise ValueError(f"Batch Process Failed: {metadata.state_message}")
+
+    storage_client = storage.Client()
+
+    # One process per Input Document
+    for process in list(metadata.individual_process_statuses):
+        # output_gcs_destination format: gs://BUCKET/PREFIX/OPERATION_NUMBER/INPUT_FILE_NUMBER/
+        # The Cloud Storage API requires the bucket name and URI prefix separately
+        matches = re.match(r"gs://(.*?)/(.*)", process.output_gcs_destination)
+        if not matches:
+            print(
+                "Could not parse output GCS destination:",
+                process.output_gcs_destination,
+            )
+            continue
+
+        output_bucket, output_prefix = matches.groups()
+
+        # Get List of Document Objects from the Output Bucket
+        output_blobs = storage_client.list_blobs(output_bucket, prefix=output_prefix)
+
+    return list(output_blobs)
+
+
+def split_document(_doc):
+    """Splits a LangChain Document into smaller chunks."""
+    # Use a recursive splitter for better semantic chunking
+    splitter = RecursiveCharacterTextSplitter(
+        chunk_size=9216,
+        chunk_overlap=200,  # Add overlap for context
+    )
+
+    new_docs = []
+    for page in _doc:
+        # Create smaller documents
+        new_chunks = splitter.create_documents([page.page_content], [page.metadata])
+
+        # Reconstruct documents with the same metadata
+        for i in range(len(new_chunks)):
+            new_chunks[i].metadata["page_chunk"] = i
+            new_chunks[i].metadata["chunk_size"] = len(new_chunks[i].page_content)
+            new_docs.append(
+                Document(
+                    page_content=new_chunks[i].page_content,
+                    metadata=new_chunks[i].metadata,
+                )
+            )
+
+    return new_docs
+
+
+# Triggered by a change in a storage bucket
+@functions_framework.cloud_event
+def process_pdf(cloud_event):
+    """Main function"""
+    data = cloud_event.data
+
+    event_id = cloud_event["id"]
+    event_type = cloud_event["type"]
+
+    bucket = data["bucket"]
+    name = data["name"]
+    metageneration = data["metageneration"]
+    timeCreated = data["timeCreated"]
+    updated = data["updated"]
+
+    print(f"Event ID: {event_id}")
+    print(f"Event type: {event_type}")
+    print(f"Bucket: {bucket}")
+    print(f"File: {name}")
+    print(f"Metageneration: {metageneration}")
+    print(f"Created: {timeCreated}")
+    print(f"Updated: {updated}")
+
+    file_name, extension = os.path.splitext(name)
+    if extension != ".pdf":
+        print("File is not a PDF. Please submit a PDF for processing instead.")
+        return
+
+    # Project vars
+    region = os.environ["REGION"]
+    project_id = os.environ["PROJECT_ID"]
+
+    # Doc AI Vars
+    source_file = "gs://{}/{}".format(bucket, name)
```

Comment:
Same as above, use f strings instead of `.format()`

Reply:
Fixed.